src/services/system.metrics.ts

```diff
@@ -0,0 +1,146 @@
+
+        const percentageCpu = 100 - ~~(100 * idleDifference / totalDifference);
+
+        resolve(percentageCpu + '%');
+      }, 100);
+    });
+  }
+
+  getMemoryUsage() {
+    const memoryUsage = process.memoryUsage();
+    return {
+      rss: memoryUsage.rss / (1024 * 1024) + ' MB',
+      heapTotal: memoryUsage.heapTotal / (1024 * 1024) + ' MB',
+      heapUsed: memoryUsage.heapUsed / (1024 * 1024) + ' MB',
+      external: memoryUsage.external / (1024 * 1024) + ' MB',
+    };
+  }
+
+  async diskUsage() {
+    const drives = os.platform() === 'win32' ? 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('').map(d => d + ':\\') : ['/'];
+
+    const results: any = [];
+
+    if (os.platform() === 'win32') {
+      const powershellPath = 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0';
+      if (!process.env.PATH?.includes(powershellPath)) {
+        process.env.PATH = `${process.env.PATH};${powershellPath}`;
+      }
+    }
+
+    for (const drive of drives) {
+      try {
+        const diskSpace = await checkDiskSpace(drive);
+        results.push({
+          drive: drive,
+          free: (diskSpace.free / (1024 * 1024 * 1024)).toFixed(2) + ' GB',
+          size: (diskSpace.size / (1024 * 1024 * 1024)).toFixed(2) + ' GB'
+        });
+      } catch (error) {
+        /* Ignore if the drive does not exist */
+      }
+    }
+
+    return results;
+  }
+
+  async startMonitoring() {
+    // const instanceId = await this.getInstanceId()    Can only be invoked inside an EC2 instance
+    const instanceId = ''
+    const { organizationId, projectId, microserviceId, serviceKey } = userAccountService.getAccountInfo();
+
+    while (true) {
+      try {
+        const cpuUsage = await this.calculateCpuUsage();
+        const memoryUsage = this.getMemoryUsage();
+        const diskUsage = await this.diskUsage();
+
+        const data = {
+          cpuUsage,
+          memoryUsage,
+          diskUsage,
+          timestamp: new Date(),
+          instanceId,
+          microserviceId
+        };
+
+        /** NOTE: commented below code for internal use only
+           metrics are saved by making axios call but commented to save cost for */
+        // if (serviceKey) {
+        //   const url = BASE_URL_SAAS + `${ORGANIZATIONS_ROUTE}/${organizationId}${PROJECTS_ROUTE}/${projectId}${SYSTEM_METRICS_ROUTE}`;
+        //   const headers = { apiKey: serviceKey }
+        //   await axiosClient.post(url, data, headers)
+
+        //   logger.trace("successfully inserted");
+        // } else {
+        await this.systemmetricsDao.create(data);
+
+        logger.trace("successfully inserted");
+        // }
+
+
+      } catch (error) {
+        console.error('Error during monitoring:', error);
+      }
+
+      /* Waiting for 1 minute before the next iteration*/
+      await new Promise(resolve => setTimeout(resolve, 60 * 1000));
+    }
+  }
+
+  async getInstanceId() {
+    try {
+      const instanceId = await axiosClient.get(EC2_METADATA_URL);
```

Comment:
take field provider in plugin: aws 
because depending on provider way to get instance id from ec2 will differ

and if provider is missing assume it it is running locally so get machine id for instanceId